src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -570,12 +571,7 @@ private void writeBuffer(ByteBuf buff, boolean end, FutureListener<Void> listene
       } else {
         l = listener;
       }
-      EventLoop eventLoop = conn.context.nettyEventLoop();
-      if (eventLoop.inEventLoop()) {
-        conn.writeBuffer(this, buff, end, l);
-      } else {
-        eventLoop.execute(() -> writeBuffer(buff, end, l));
-      }
+      conn.writeBuffer(this, buff, end, l);
```

Comment:
we need to actually keep if/else event loop block because we really want the method `writeBuffer` to always be executed on the event-loop and not be concurrency